src/_document/getDocumentInitialProps.tsx

```diff
@@ -0,0 +1,24 @@
+import { Fragment } from 'react';
+import NextDocument, { DocumentInitialProps } from 'next/document';
+import type { PageObject } from '../commonTypes';
+import type { DocumentType, RenderPage } from 'next/dist/next-server/lib/utils';
+
+export default async function getDocumentInitialProps({
+  Document,
+  pageObject,
+  renderPage,
+}: {
+  Document: DocumentType;
+  pageObject: PageObject;
+  renderPage: RenderPage;
+}): Promise<DocumentInitialProps> {
+  const getDocumentInitialProps =
+    Document.getInitialProps || NextDocument.getInitialProps;
```

Comment:
Do we need this fallback to `NextDocument.getInitialProps` ? If I'm not mistaken custom Document object should extend Next.js `Document` which is the same class we use here as fallback.

Reply:
I think `getInitialProps` is typed as optional in type definitions, so we do a fallback to avoid having to do checks.

Reply:
👍 It seems that the fallback never happen on runtime, but TS complains about it. Skipped that line for code coverage.